test/tools/unified-spec-runner/match.ts

```diff
@@ -241,13 +241,13 @@ export function resultCheck(
       }
       for (const [index, value] of expectedEntries) {
         path.push(`[${index}]`);
-        checkNestedDocuments(index, value, false);
+        checkNestedDocuments(index, value, checkExtraKeys);
```

Comment:
For these we pass the `checkExtraKeys` value through all the way - no more hardcoded assumptions.